doc/Core.xml

```diff
@@ -2188,7 +2190,61 @@ DATE: when response was generated
         </figure>
         <para>Both digest authentication and the user name token profile give only a rudimentary level of security. In a system where security is important, it is recommended to always configure the device for TLS-based access (see Advanced Security Service). Digest authentication or the user name token message level security combined with TLS, with client and server authentication, protected transport level security give an acceptable level of security in many systems.</para>
         <para>An ONVIF compliant device should authenticate an RTSP request at the RTSP level. If HTTP is used to tunnel the RTSP request the device shall not authenticate on the HTTP level.</para>
-        <para>When authenticating RTSP or HTTP methods, an ONVIF compliant device shall use digest authentication [RFC 2617]. The credentials shall be managed with the GetUsers, CreateUsers, DeleteUsers and SetUser methods. If the device also supports WS-Security, the same set of credentials shall be used.</para>
+        <para>When authenticating RTSP or HTTP methods, an ONVIF compliant device shall use digest authentication [RFC 2617]and [RFC 7616. The credentials shall be managed with the GetUsers, CreateUsers, DeleteUsers and SetUser methods. If the device also supports WS-Security, the same set of credentials shall be used.</para>
+      </section>
+	  <section>
+	  <title>GetHashingAlgorithm</title>
+        <section>          
+          <para>ONVIF Client gets current hashing algorithm of ONVIF device by sending ONVIF HTTP or RTSP request with out Authorization header.Here ONVIF Device responds with a HTTP or RTSP 401 message according to [RFC 2617] [RFC 7616] that includes a challenge for each digest algorithm currently it was set, in its order of preference.</para>
+          <para>The example HTTP digest challenge response from ONVIF Device when ONVIF Device current hashing algorithams is MD5 and SHA-256 is as follows.</para>
+		  <programlisting><![CDATA[HTTP/1.1 401 Unauthorized
+				WWW-Authenticate: Digest
+					realm="http-auth@test.org",
+					qop="auth, auth-int",
+					algorithm=MD5,
+					nonce="7ypf/xlj9XXwfDPEoM4URrv/xwf94BcCAzFZH4GiTo0v",
+					opaque="FQhe/qaU925kfnzjCev0ciny7QMkPqMAFRtzCUYo5tdS"
+				WWW-Authenticate: Digest
+					realm="http-auth@test.org",
+					qop="auth, auth-int",
+					algorithm=SHA-256,
+					nonce="7ypf/xlj9XXwfDPEoM4URrv/xwf94BcCAzFZH4GiTo0v",
+					opaque="FQhe/qaU925kfnzjCev0ciny7QMkPqMAFRtzCUYo5tdS"]]>
+			</programlisting>
+			<para>To maintain backward compatability with old ONVIF Client(s), ONVIF Device by default should do HTTP or RTSP digest challenge with MD5 hashing alogorithm as per RFC[2617].</para>
+        </section>
+      </section>
```

Comment:
Please check the attached image.
empty heading and  REQUEST/RESPONSE section missing.

getservicecapability section also needs update.

![digest](https://user-images.githubusercontent.com/67031640/180144724-628b7104-f864-463c-9a16-3a9ca29915bd.png)

Reply:
thanks sujith, let me review it.

Reply:
updated PR by incorporating review comments.

Reply:
updated PR as per review comments 
a. default  device behavior. 
capability check.
c. how to get hashing algorithm
d. how to get current hashing algorithm.